connector_importer/components/importer.py

```diff
@@ -0,0 +1,355 @@
+# Author: Simone Orsi
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields
+from odoo.addons.component.core import Component
+from ..log import logger
+from ..log import LOGGER_NAME
+
+
+class RecordSetImporter(Component):
+    """Importer for recordsets."""
+
+    _name = 'importer.recordset'
+    _inherit = 'importer.base.component'
+    _usage = 'recordset.importer'
+    _apply_on = 'import.recordset'
+
+    def run(self, recordset, **kw):
+        """Run recordset job.
+
+        Steps:
+
+        * update last start date on recordset
+        * read source
+        * process all source lines in chunks
+        * create an import record per each chunk
+        * schedule import for each record
+        """
+        # update recordset report
+        recordset.set_report({
+            '_last_start': fields.Datetime.now(),
+        }, reset=True)
+        msg = 'START RECORDSET {0}({1})'.format(recordset.name,
+                                                recordset.id)
+        logger.info(msg)
+
+        source = recordset.get_source()
+        for chunk in source.get_lines():
+            # create chuncked records and run their imports
+            record = self.env['import.record'].create(
+                {'recordset_id': recordset.id})
+            # store data
+            record.set_data(chunk)
+            record.run_import()
+
+
+class RecordImporter(Component):
+    """Importer for records.
+
+    This importer is actually the one that does the real import work.
+    It loads each import records and tries to import them
+    and keep tracks of errored, skipped, etc.
+    See `run` method for detailed information on what it does.
+    """
+
+    _name = 'importer.record'
```

Comment:
imo, this is an extremely confusing name with `import.record`. Maybe it's better to follow general naming and call it `importer.importer` or smth

Reply:
this is the importer of the record.
The distinction overall, is this:
`import.` identifies odoo records
`impoter.` identifies what has to do w/ importers (importers, mappers, etc)

Reply:
worth to mention: I spent a bit of time in finding the right way to name those things... I never came up w/ a good solution w/out messing up w/ extra super-long names :cry: 
But I'm open to suggestions :wink: